### PR TITLE
Add Support for "Blank" Link components.

### PR DESCRIPTION
**Pull request by annarankin**
Hello, folks!

This PR addresses Issue https://github.com/reactjs/react-router/issues/2319. It allows Link components to render anchor tags without `href`s when the `to` prop is not specified.

Love to hear what you think - I hope this is helpful! :v:

### CODE REVIEW

File: examples/empty-links/app.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+)
+
+render((
+  <Router history={withExampleBasename(browserHistory, __dirname)}>
+    <Route path="/" component={App}>
+      <IndexRoute component={Index}/>
+      <Route path="/somewhere" component={Somewhere}/>
+    </Route>
+  </Router>
+), document.getElementById('example'))
```
Comment (timdorr) [inline examples/empty-links/app.js]: I don't think this particular edge case needs an example for it. Please remove it.

File: modules/Link.js [hunk 2]
```diff
@@ -112,6 +113,9 @@ const Link = React.createClass({
     const { router } = this.context
 
     if (router) {
+      // If user does not specify a `to` prop, return an empty anchor tag.
+      if (to === '') { return <a {...props } /> }
```
Comment (taion) [inline modules/Link.js]: Just use `== null`. No need to set a default here I think?

File: modules/__tests__/Link-test.js [hunk 3]
```diff
@@ -455,4 +455,45 @@ describe('A <Link>', function () {
...
+
+    it('returns an anchor tag without an href', function (done) {
+      render((
+        <Router history={createHistory('/')}>
+          <Route path="/" component={App} />
+        </Router>
+      ), node, function () {
+        const link1 = node.querySelectorAll('a')[0]
+        const link2 = node.querySelectorAll('a')[1]
+        const link3 = node.querySelectorAll('a')[1]
```
Comment (timdorr) [inline modules/__tests__/Link-test.js]: Should be [2]

**Comment by taion** [comment]
LGTM

**Pull request merged**